templates/static-files.conf

```diff
@@ -0,0 +1,60 @@
+server {
+        listen 80;
+        server_name {{ site_url }};
+
+        {% if use_ssl %}
+        return 301 https://{{ site_url }}$request_uri;
+        {% else %}
+        root {{ site_path }};
+
+        location / {
+                index index.html;
+        }
+
+        include /etc/nginx/locations.d/{{ site_url }}/*.conf;
+        {% endif %}
+}
+
+{% if use_ssl %}
+server {
+        listen 443 ssl http2;
+        listen [::]:443 ssl http2;
+        server_name {{ site_url }};
+
+        # certs sent to the client in SERVER HELLO are concatenated in ssl_certificate
+        ssl_certificate {{ ssl_certificate_path }};
+        ssl_certificate_key {{ ssl_key_path }};
+        ssl_session_timeout 1d;
+        ssl_session_cache shared:SSL:50m;
+        ssl_session_tickets off;
+
+        # Diffie-Hellman parameter for DHE ciphersuites, recommended 2048 bits
+        ssl_dhparam /etc/nginx/dhparam.pem;
+
+        # modern configuration. tweak to your needs.
+        ssl_protocols TLSv1.2;
+        ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+        ssl_prefer_server_ciphers on;
+
+        # HSTS (ngx_http_headers_module is required) (15768000 seconds = 6 months)
+        add_header Strict-Transport-Security max-age=15768000;
+
+        # OCSP Stapling ---
+        # fetch OCSP records from URL in ssl_certificate and cache them
+        ssl_stapling on;
+        ssl_stapling_verify on;
+
+        ## verify chain of trust of OCSP response using Root CA and Intermediate certs
+        ssl_trusted_certificate /etc/ssl/cert.pem;
+
+        resolver {{ ssl_resolver }};
```

Comment:
Not really sure where this gets set, I think at the moment its a setting the user must set by hand.  It should default via the defaults/main.yml file and ideally should point to a resolver that is always up.  I like 8.8.8.8 but I believe you also need the v6 resolver if the client is requesting via a v6 path.  Perhaps this can be specified multiple times or take a list of resolvers?

Either way this also needs an associated firewall rule.